### ISSUE TITLE
import/named doesn't work when the target has no exports

### ISSUE THREAD

**Issue by jf248** [benmosher/eslint-plugin-import#1147]
Given:
```
// a.js
console.log('This file has no exports')

// b.js
import { doesntExist } from './a'
```
I'd expect the `import/named` rule to show an error for `doesntExist`. It doesn't. What am I missing? Is there a way to check for this type of error?

**Comment by ljqx** [comment]
This is because the `eslint-module-utils` wrongly consider file without `import` or `export` as 
> definitely _not_ a module

https://github.com/benmosher/eslint-plugin-import/blob/798eed7e559adab2eac07bf1b3e3518d4b7a4296/utils/unambiguous.js#L5

But this is not true. Module could have no `import` or `export`.

The test covers one case that make sense, that file with CommonJS require/exports.

But as there is another feature request that to check CommonJS files too #1145, and there is already one PR for it #1226, it's better to wait for that to be completed to start to fix this one.

**Comment by ljharb** [comment]
Indeed, there's no possible way to know if a file without imports, exports, or `with` is a Module or a Script - the parsing goals are ambiguous.
